applications/vanilla/views/discussion/helper_functions.php

```diff
@@ -454,6 +488,14 @@ function getCommentOptions($comment) {
                 'Url' => '/post/editcomment/'.$comment->CommentID,
                 'EditComment'
             ];
+
+            if (!empty(val('DateUpdated', $comment))) {
```

Comment:
Same kinda permission issue with the discussion link. A user with the ability to edit their post will see the "Revision History" link, but it'll only lead them to a permission error page.